src/self-binding-behavior.js

```diff
@@ -0,0 +1,22 @@
+function findOriginalEventTarget(event) {
+  return (event.path && event.path[0]) || (event.deepPath && event.deepPath[0]) || event.target;
+}
+
+function handleSelfEvent(event) {
+  let target = findOriginalEventTarget(event);
+  if (this.target !== target) return;
+  this.originalCallSource(event);
+}
+
+export class SelfBindingBehavior {
+  bind(binding, source) {
+    if (!binding.callSource || !binding.targetEvent) throw new Error('Self binding behavior only supports event.');
+    binding.originalCallSource = binding.callSource;
```

Comment:
instead of `originalCallSource` (which might conflict with someone else's binding behavior logic) try using `selfCallSource` or something else that is specific to this binding behavior.